src/main/java/org/commcare/formplayer/util/RequestUtils.java

```diff
@@ -57,18 +62,19 @@ public static String getBody(HttpServletRequest request) throws IOException {
         return body;
     }
 
-    public static JSONObject getPostData(FormplayerHttpRequest request) {
+    public static JSONObject getPostData(HttpServletRequest request) {
         JSONObject data = null;
         try {
             data = new JSONObject(getBody(request));
         } catch (IOException | JSONException a) {
-            throw new RuntimeException("Unreadable POST Body for the request: " + request.getRequestURI());
+            throw new RuntimeException("Unreadable POST Body for the request: " + request.getRequestURI(), a);
         }
         return data;
     }
 
-    public static String getRequestEndpoint(HttpServletRequest request) {
-        return StringUtils.strip(request.getRequestURI(), "/");
+    public static String getRequestEndpoint() {
+        HttpServletRequest request = RequestUtils.getCurrentRequest();
+        return request == null ? "unknown" : StringUtils.strip(request.getRequestURI(), "/");
```

Comment:
Is this method just informational / user facing? It seems like this is potentially dangerous to infer a real string from

Reply:
this value is only used for metric tag values. What do you mean about "infering a real string"?

Reply:
All good if it's just metric tags.

If this was, say, used for request routing, I was concerned that there might be a scenario where providing a real value (`unknown`) when something is missing could be a sidechannel into routing if there was a real potential `unknown`  output from the call.